fitapp/tests/test_retrieval.py

```diff
@@ -179,20 +179,37 @@ def test_subscription_update_locked(self, mock_add, get_fitbit_data):
         self.assertEqual(TimeSeriesData.objects.count(), 0)
 
     @patch('fitapp.utils.get_fitbit_data')
-    @patch('fitapp.tasks.get_time_series_data.retry')
-    def test_subscription_update_too_many(self, mock_retry, get_fitbit_data):
+    def test_subscription_update_too_many(self, get_fitbit_data):
         # Check that celery tasks get postponed if the rate limit is hit
-        mock_retry.return_value = celery.exceptions.Retry()
+        cat_id = getattr(TimeSeriesDataType, self.category)
+        _type = TimeSeriesDataType.objects.filter(category=cat_id)[0]
+        lock_id = 'fitapp.tasks-lock-{0}-{1}-{2}'.format(
+            self.fbuser.fitbit_user, _type, self.date)
         exc = fitbit_exceptions.HTTPTooManyRequests(self._error_response())
         exc.retry_after_secs = 21
-        get_fitbit_data.side_effect = exc
+        def side_effect(*args, **kwargs):
+            # Delete the cache lock after the first try and adjust the
+            # get_fitbit_data mock to be successful
+            cache.delete(lock_id)
+            get_fitbit_data.side_effect = None
+            get_fitbit_data.return_value = [{
+                'dateTime': self.date,
+                'value': '34'
+            }]
+            raise exc
+        get_fitbit_data.side_effect = side_effect
```

Comment:
@brad Let me see if I understand the code. It seems that we are setting `get_fitbit_data.side_effect` to a function but inside the same function we set the same attribute `side_effect` to `None` ? Is that how it works?